lib/Common/DataStructures/ImmutableList.h

```diff
@@ -729,7 +731,9 @@ namespace regex
                 return this;
             }
             auto arr = ToReferenceArrayInHeap(size);
-            QuickSort<T *, true>::Sort(arr, arr+(size-1), comparer);
+            qsort_s(arr, size, sizeof(T*), [](void* comparer, const void* a, const void* b) {
```

Comment:
I fail to see what makes this sort in reverse order, but this refactor shouldn't  change the current behavior.